lib/galaxy/tool_util/linters/inputs.py

```diff
@@ -32,19 +32,21 @@ def lint_inputs(tool_xml, lint_ctx):
         if param_type == "data":
             if "format" not in param_attrib:
                 lint_ctx.warn("Param input [%s] with no format specified - 'data' format will be assumed.", param_name)
-
-        if param_type == "select":
+        elif param_type == "select":
             dynamic_options = param.get("dynamic_options", None)
             if dynamic_options is None:
                 dynamic_options = param.find("options")
 
             select_options = _find_with_attribute(param, 'option', 'value')
```

Comment:
There was a copy-paste error here since this code was introduced in commit 630ac3658f3c64bd9a9abe5757cce0ed01cad229 , this will never find select options without a value.

```suggestion
            select_options = param.findall('./option')
```